contracts/src/v0.8/ccip/test/pools/GHO/GhoTokenPoolEthereum.t.sol

```diff
@@ -0,0 +1,996 @@
+        remoteChainSelector: SOURCE_CHAIN_SELECTOR,
+        receiver: OWNER,
+        localToken: address(s_token),
+        sourcePoolAddress: abi.encode(s_sourcePool),
+        sourcePoolData: bytes(""),
+        offchainTokenData: bytes("")
+      })
+    );
+
+    assertEq(s_ghoTokenPool.getCurrentBridgedAmount(), 0);
+  }
+
+  function testFuzz_ReleaseOrMintSuccess(address recipient, uint256 amount, uint256 bridgedAmount) public {
+    // Since the owner already has tokens this would break the checks
+    vm.assume(recipient != OWNER);
+    vm.assume(recipient != address(0));
+    vm.assume(recipient != address(s_token));
+
+    amount = uint128(bound(amount, 2, type(uint128).max));
+    bridgedAmount = uint128(bound(bridgedAmount, amount, type(uint128).max));
+
+    // Inflate current bridged amount so it can be reduced in `releaseOrMint` function
+    vm.startPrank(AAVE_DAO);
+    s_ghoTokenPool.setBridgeLimit(bridgedAmount);
+    s_ghoTokenPool.setChainRateLimiterConfig(
+      DEST_CHAIN_SELECTOR,
+      RateLimiter.Config({isEnabled: true, capacity: type(uint128).max, rate: 1e15}),
+      RateLimiter.Config({isEnabled: true, capacity: type(uint128).max, rate: 1e15})
+    );
+    vm.warp(block.timestamp + 1e50); // wait to refill capacity
+    vm.startPrank(s_allowedOnRamp);
+    s_ghoTokenPool.lockOrBurn(
+      Pool.LockOrBurnInV1({
+        receiver: bytes(""),
+        remoteChainSelector: DEST_CHAIN_SELECTOR,
+        originalSender: STRANGER,
+        amount: bridgedAmount,
+        localToken: address(s_token)
+      })
+    );
+
+    // Makes sure the pool always has enough funds
+    deal(address(s_token), address(s_ghoTokenPool), amount);
+    vm.startPrank(s_allowedOffRamp);
+
+    uint256 capacity = _getInboundRateLimiterConfig().capacity;
+    uint256 bridgedAmountAfter = bridgedAmount;
+    // Determine if we hit the rate limit or the txs should succeed.
+    if (amount > capacity) {
+      vm.expectRevert(
+        abi.encodeWithSelector(RateLimiter.TokenMaxCapacityExceeded.selector, capacity, amount, address(s_token))
+      );
+    } else {
+      // Only rate limit if the amount is >0
+      if (amount > 0) {
+        vm.expectEmit();
+        emit TokensConsumed(amount);
+      }
+
+      vm.expectEmit();
+      emit Released(s_allowedOffRamp, recipient, amount);
+
+      bridgedAmountAfter -= amount;
+    }
+
+    s_ghoTokenPool.releaseOrMint(
+      Pool.ReleaseOrMintInV1({
+        originalSender: bytes(""),
+        amount: amount,
+        remoteChainSelector: SOURCE_CHAIN_SELECTOR,
+        receiver: recipient,
+        localToken: address(s_token),
+        sourcePoolAddress: abi.encode(s_sourcePool),
+        sourcePoolData: bytes(""),
+        offchainTokenData: bytes("")
+      })
+    );
+
+    assertEq(s_ghoTokenPool.getCurrentBridgedAmount(), bridgedAmountAfter);
+  }
+
+  function testChainNotAllowedReverts() public {
+    vm.startPrank(s_allowedOnRamp);
+    s_ghoTokenPool.lockOrBurn(
```

Comment:
this is prepping the scenario right? can we add a comment?

Reply:
yes, it's to increase `currentBridgedAmount` such the later offRamp doesn't prematurely revert with `NotEnoughBridged()` 
was written when we had not settled on adding the setter which i will now use haha

Reply:
[f393191](https://github.com/aave/ccip/pull/20/commits/f3931913517e49c57bb19539802c2e3123137132)